server/workflow/workflow_server.go

```diff
@@ -171,8 +169,12 @@ func (s *workflowServer) WatchWorkflows(req *workflowpkg.WatchWorkflowsRequest,
 		select {
 		case <-ctx.Done():
 			return nil
-		case event, open := <-watch.ResultChan():
-			if !open {
+		case event, ok := <-watch.ResultChan():
+			var wf *wfv1.Workflow
+			if ok {
+				wf, ok = event.Object.(*wfv1.Workflow)
+			}
+			if !ok {
```

Comment:
Could you explain this? Not sure how this fixes the issue?

Reply:
essentially, I'm assuming if the result channel is closed, or the object is not an workflow (I'm assuming that this is a `*api.Status` error), but I don't think we care what it is, we just want to re-establish:

```golang
type Event struct {
	Type EventType

	// Object is:
	//  * If Type is Added or Modified: the new state of the object.
	//  * If Type is Deleted: the state of the object immediately before deletion.
	//  * If Type is Bookmark: the object (instance of a type being watched) where
	//    only ResourceVersion field is set. On successful restart of watch from a
	//    bookmark resourceVersion, client is guaranteed to not get repeat event
	//    nor miss any events.
	//  * If Type is Error: *api.Status is recommended; other types may make sense
	//    depending on context.
	Object runtime.Object
}
```